### ISSUE TITLE
Should we merge the M23 and M35 classes?

### ISSUE THREAD

**Issue by pasqLisena** [DOREMUS-ANR/doremus-ontology#10]
The class `M23 Casting Detail` is a part of a particular casting. The properties that can be assigned to its instance are:

- U36 foresees responsibility of type M32 Actor’s Responsibility (_interprète_, _soliste_);
- U9 specifies character the character F28 Character (_Carmen_),
- U28 has intended tessitura M30 Tessitura (from the lower to the higher tone)
- U1 has intended MOP **M35 Cast Medium of Performance**

The class `M35 Cast Medium of Performance` specifies:
- U39 has medium of performance M14 Medium of Performance (piano, sax, orchestra, voice, ...)
- P90 has value E60 Number, that is the number of that specific MOP (1 piano, 2 clarinets, ...).

This is an [example](https://drive.google.com/file/d/0B4s5dBwAmhJscWszT2hTZEdPSU0/view?usp=sharing) of the way we use these classes currently:

<img width="673" alt="schermata 2016-08-19 alle 16 07 04" src="https://cloud.githubusercontent.com/assets/4126222/17812365/088fe436-6627-11e6-9e12-fec986e45861.png">

In all examples, for each M23 there is exactly one M35.

**My proposal** is to merge the two classes in a unique one (M23), with these properties:
- U36 foresees responsibility of type M32 Actor’s Responsibility
- U9 specifies character the character F28 Character 
- U28 has intended tessitura M30 Tessitura 
- U39 has medium of performance M14 Medium of Performance 
- P90 has value E60 Number

**Comment by pierrechoffe** [comment]
@pasqLisena Hi from Mill Valley, California :))

I totally agree with your point of view, this has been a pain in the back since the beginning. The problem is that in FRBRoo (as in CRM) there is only one way to quantify sth, it is through E54 Dimension. This was confirmed to me once again at the latest CRM SIG meeting in August. I see it as a limitation, maybe there is a simpler RDF solution?

Meanwhile, the E54 Dimension is used to say that we have an E58 Measurement Unit (in this case the unit is a piano for instance) and an E60 Number (e.g. 1)
M35 Cast MOP is a subclass of E54 Dimension, so this is a dimension. M14 MOP is a subclass of E55 Type *and* of E58 Measurement Unit.

The reason why we didn't merge M35 and M23 is that in my opinion it makes no sense to say that a dimension has a quality of soloist or specifies a character.

Ideally, if we can find a way of assigning a quantity without using the concept of Dimension, we could have a unique M23 Casting Detail (pretty much like your proposal) with : 

* has intended quantity "1"
* has intended mop "piano"
* has intended responsibility type "soloist"

Careful : 

(a) some properties are deprecated (u9) or changed scope (u28)
(b) the new way of describing characters in an F22 is described here : 

[characters.pdf](https://github.com/DOREMUS-ANR/doremus-ontology/files/427494/characters.pdf)

**Comment by pasqLisena** [comment]
> maybe there is a simpler RDF solution?

For sure, @rtroncy knows the answer to this.

> Ideally, if we can find a way of assigning a quantity without using the concept of Dimension, we could have a unique M23 Casting Detail (pretty much like your proposal) with :
> 
> has intended quantity "1"
> has intended mop "piano"
> has intended responsibility type "soloist"

Yes, this is ideally the best solution. But you missed the tessitura?

> Careful :
> (a) some properties are deprecated (u9) or changed scope (u28)
> (b) the new way of describing characters in an F22 is described here :

Ok, but I have no evidence of this in the ontology here on GitHub. I suppose that these edits are in progress, so when they will became definitive (and committed 😄 ) it will not be a problem align everything.

**Comment by pierrechoffe** [comment]
@pasqLisena you can check the work in progress on Google Drive, Modelisation/classes & properties : 
https://docs.google.com/spreadsheets/d/1RNhHnp4fWEhG8G4acx8FPi-nu9fUWNhc8JNf3SQwUFE/edit#gid=1401336326
(see tab "propriétés")
column "protégé" : if the property is in Protégé, then "ok". If the property was created recently, then the protégé column is void.

Regarding "tessitura" : following the new modelling of characters, the tessitura will be modified. As a matter of fact, tessitura is attached to the Character, not to the Casting Detail.

I am eager to know @rtroncy 's RDF solution to our quantity problem 😄

**Comment by rtroncy** [comment]
We don't need the concept of Dimension as in the CRM sense. Counting objects can be done with a simple property. Consequently, we don't need anymore the class M35.

**Comment by pierrechoffe** [comment]
Greaaaat ! how would this work?

**Comment by pasqLisena** [comment]
I cite @rtroncy in #11:

> There is no RDF built-in property for counting things if this is your question. But, nothing prevent us to create a property for this. Consequently, the `p90_has_value` is certainly not the only way of expressing this information and this specific property will be quite badly labeled for this purpose.

**Comment by rtroncy** [comment]
Let me clarify a bit the issue: using the concept of Dimension (as in the CRM sense) is useful when one wants to say more things than just indicating a number of objects. For example, schema.org defines the concept of [StructuredValue](https://schema.org/StructuredValue) and more specific types such as [QuantitativeValue](https://schema.org/QuantitativeValue) or [PropertyValue](https://schema.org/PropertyValue) which are analogous to a E54, but in this case, we can attach more information such as a min and a max value, code and units, etc.

If our requirement amounts to just count the number of MOP in a casting, then, a simple property can be used. Concretely, a full example would be:

```
<http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7>
  a efrbroo:F22_Self-Contained_Expression ;
  mus:U13_has_intended_casting <http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7/casting/1> .

<http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7/casting/1>
  a mus:M6_Intended_Casting ;
  ecrm:P3_has_note "Piano, Alto" ;
  mus:U23_has_casting_detail <http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7/casting/1/detail/1> ;
  mus:U23_has_casting_detail <http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7/casting/1/detail/2> .

<http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7/casting/1/detail/1>
  a mus:M23_Casting_Detail ;
  mus:U1_has_intended_medium_of_performance <http://iflastandards.info/ns/unimarc/terms/mop/kpf> ;
  mus:Uxx_has_number_medium_of_performance 1^^xsd:int ;
  mus:U36_foresees_responsibility_of_type mus:Ixx_Soliste .

<http://data.doremus.org/expression/10430039-5e02-3ec8-bc6b-b433cdde13c7/casting/1/detail/2>
  a mus:M23_Casting_Detail ;
  mus:U1_has_intended_medium_of_performance <http://iflastandards.info/ns/unimarc/terms/mop/val> ;
  mus:Uxx_has_number_medium_of_performance 1^^xsd:int .
```

Further, I would propose to rename the following classes and properties:
* M6_Intended_Casting :arrow_right: M6_Casting
* U13_has_intended_casting :arrow_right: U13_has_casting
* U1_has_intended_medium_of_performance :arrow_right: U1_has_medium_of_performance
I know that you want to distinguish what is an intended casting versus what casting will actually perform during a performance, but AFAIK, you haven't created yet those non-intended properties which is anyway not useful since the distinction will already be made by virtue of being attached to a performance.

I would also recommend to keep M23_Casting_Detail instead of M23_Detailed_Casting as you did in the [spreadsheet](https://docs.google.com/spreadsheets/d/1RNhHnp4fWEhG8G4acx8FPi-nu9fUWNhc8JNf3SQwUFE/edit#gid=317279816), the former being clearer, this is NOT a different type of casting but a part of a casting.

I would recommend to delete the class M32_Actor_Responsibility and instead, create 3 named individuals in OWL corresponding to the values: soliste, choriste, interprète musical. I have tentatively identified them by Ixx_ but they could also well be Mxx_.

@pierrechoffe Can you explain how the new U28_has_intended_tessitura property works with a concrete example? To what object is this property attached? I would also make a similar remark for the name: why not U28_has_tessitura?

Final remark: the solution above includes the definition of a new Uxx_has_number_medium_of_performance property. This will just enable to count MOP. If we need to count other type of objects, we may want to generalize this property into a simpler Uxx_has_number

**Comment by pierrechoffe** [comment]
@rtroncy thanks for this !
How would we define our new property? what would be its superproperty?
(i'm talking about the line : Uxx_has_number_medium_of_performance 1^^xsd:int)

**Comment by pierrechoffe** [comment]
@rtroncy we can remove the "intended" in the property names. This was supposed to clarify things but it may be heavy.

>I would also recommend to keep M23_Casting_Detail instead of M23_Detailed_Casting as you did in the spreadsheet, the former being clearer, this is NOT a different type of casting but a part of a casting.

Totally agree with you, the latter is an older version of the class name and it reappears sometimes to create confusion...

>Uxx_has_number_medium_of_performance property. This will just enable to count MOP. If we need to count other type of objects, we may want to generalize this property into a simpler Uxx_has_number

We may have to count the number of "intended" interpreters (e.g. in a 4-hand piano there are 2 interpreters).

>I would recommend to not directly use the class M32_Actor_Responsibility and instead, create 3 named individuals in OWL corresponding to the values: soliste, choriste, interprète musical.

We may need more than these 3 values. This has to be confirmed by the modeling group.

**Comment by rtroncy** [comment]
> How would we define our new property? what would be its superproperty?

```Uxx_has_number_medium_of_performance``` will be a simple ```owl:DataProperty``` which has for range a ```xsd:nonNegativeInteger```. There is no need to have a superproperty.

> We may have to count the number of "intended" interpreters (e.g. in a 4-hand piano there are 2 interpreters).

What are interpreters in DOREMUS? Are they not MOP?

> We may need more than these 3 values. This has to be confirmed by the modeling group.

The documentation just mentions those values, so if this is not exhaustive, we need to know. We can always add new OWL individuals but if the enumeration becomes large, then we better rely on one more SKOS controlled vocabulary, this is the trade-off we are following so far.

**Comment by pasqLisena** [comment]
>  > We may have to count the number of "intended" interpreters (e.g. in a 4-hand piano there are 2 interpreters).

> What are interpreters in DOREMUS? Are they not MOP?

In that specific case there are 2 pianists but 1 piano. But we currently do not represent them in Doremus, so I suggest to not consider it for now.

**Comment by pierrechoffe** [comment]
@rtroncy 
>Can you explain how the new U28_has_intended_tessitura property works with a concrete example? To what object is this property attached? 

See the attached schema. Note that we can drop all the "intended" in the property/class names.

[characters 2.pdf](https://github.com/DOREMUS-ANR/doremus-ontology/files/451598/characters.2.pdf)

**Comment by pierrechoffe** [comment]
@rtroncy I am presently working on the new Casting schema. The M23 Casting Detail would remain as-is, i.e. a subclass of E29 Design or Procedure. I have a question : 

I already defined properties for the M27 Foreseen Individual Performance (also a subclass of E29) that could be used for the "intended" casting, e.g. u2_foresees_use_of_medium_of_performance or u36_foresees_responsibility_of_type, but the meaning of "foresees" is not the same as "intended". "foresees" means that a previsional event is supposed to be performed so & so by the performers (e.g. performing on a modern Steinway piano). "intended" means that the composer had this specific intention when he composed (e.g. that the music should be performed on a harpsichord).

Functionally,  we could just as well keep only one property. We just need to have a high-level domain (E29) instead of a narrow one (M27 or M23). So we could keep the u2 and u36 properties not only for M27 Foreseen Individual Performance, but also for the M23 Casting Detail. But doesn't it make things clearer to have a specific property for each case?

**Comment by pasqLisena** [comment]
Hi Pierre,
we can take also in consideration designing `M27 Foreseen Individual Performance` as a subclass of `M23 Casting Detail` (the difference between the 2 is in describing details about the instrument and the performer).

**Comment by pierrechoffe** [comment]
Hi Pasquale,

While this may work from a technical point of view, since both are Designs or Procedures, remember : 

(1) M27 is a subclass of M26_Foreseen_Performance which is a subclass of M25_Foreseen_Activity
(2) we want a clear separation between "notated music" (the composition) and "performed music" (the interpretation).
(3) conceptually, a Casting (and a Casting Detail) have nothing to do with a performance (be it "foreseen").
So I would not favour this solution.

My proposal was to use the same properties, but not to create a hierarchy between these classes. They may have the same ancestor (E29) but they represent totally different concepts.

Actually, my question regarding using or not the same properties was a follow-up of our discussion with @rtroncy  : 
>Further, I would propose to rename the following classes and properties ("intended"...)
I know that you want to distinguish what is an intended casting versus what casting will actually perform during a performance, but AFAIK, you haven't created yet those non-intended properties which is anyway not useful since the distinction will already be made by virtue of being attached to a performance.

If we drop the notion of "intended", we might as well use the same properties in both cases ("notated music" and "performed music") just as here : https://docs.google.com/spreadsheets/d/10Oe34ZSIl3bJEVO8O1pq_NoTUuEkZpU3yd5-UMes9vs/edit#gid=0

In this spreadsheet, I would suggest to have in "notated music" the same properties as in "foreseen performance" (except for specific object). We could delete u52 and u1.

What do you think ?

**Comment by pasqLisena** [comment]
> conceptually, a Casting (and a Casting Detail) have nothing to do with a performance (be it "foreseen")

I am thinking about it differently. In a nutshell, the Casting is the way in which the composer think its work should be performed.
From this point of view, both Casting and Foreseen Performance are potential Performances.

You can also see it in the reverse (so Casting being a subclass of Foreseen Performance).

> In this spreadsheet, I would suggest to have in "notated music" the same properties as in "foreseen performance" (except for specific object). We could delete u52 and u1.

I agree. In addiction, I think that we can use the same property for the 3 (i.e. a unique `has medium of performance`).

The difference between the intention, the foresight and the actually performance is not in the mop, but in the performance/casting itself. In other words, we can distingue them only at the class level, while we let them share these properties.

**Comment by pierrechoffe** [comment]
>I agree. In addiction, I think that we can use the same property for the 3 (i.e. a unique has medium of performance).

We cannot use the same property for the 3 because in the case of the real performance the domain is an event, whereas for the 2 foreseen the domain is a conceptual object.

**Comment by pierrechoffe** [comment]
>You can also see it in the reverse (so Casting being a subclass of Foreseen Performance).

Yes, as a subclass of M26_Foreseen_Performance it makes more sense than as a subclass of M27. Yet...  we could say the same about the F25 Performance Plan, which is a subclass of E29. I would rather keep M6 and M23 at the level of F25, i.e. s subclasses of E29.

**Comment by rtroncy** [comment]
Guys, while this discussion is very interesting, the idea of github issues is to discuss focus and narrow bugs that can be addressed, not the complete model in one issue. Are you still discussing the merge of the M23 and M35? If not, then open a new issue

**Comment by pierrechoffe** [comment]
@pasqLisena @rtroncy I think we agree on the essential, i.e. supressing M35 and describing the MOP (quality & quantity) at the level of M23. The RDF would be identical to your full example, with small changes on the properties.
So, if Pasquale agrees, we can close this issue.

**Comment by rtroncy** [comment]
We will close it when the ontology will be modified :-) For the rest of what you're discussing, is there a need to open new issues?

**Comment by pierrechoffe** [comment]
Not for me. Pasquale ?

**Comment by pasqLisena** [comment]
Ok for me, we close the issue in the next commit.

**Comment by rtroncy** [comment]
> Not for me. 

I'm surprised @pierrechoffe that there is no need for a longer discussion. My impression is that this idea of duplicating several properties to distinguish a foreseen performance versus an actual performance that took place, adding as other dimension also what is written in notated music is very confusing. I look also at this  [spreadsheet](https://docs.google.com/spreadsheets/d/10Oe34ZSIl3bJEVO8O1pq_NoTUuEkZpU3yd5-UMes9vs/edit#gid=0). I think this should deserve a proper new issue.

Before going further, you should really ask yourself: what fundamentally is different in the bare definition of a u2 and a u18?, between a u35 and a u31?, etc. You cannot say: this is the difference between what was planned versus what occurred ... because you get this with the broader graph (all the other property / values). So I'm really puzzled ...

**Comment by pierrechoffe** [comment]
@rtroncy I updated the spreadsheet to make things clearer. 
Column B is what we have so far (for M23) and column C is my proposal.
As you can see my proposal is to use the same properties in the "foreseen performance" and in the "casting detail".
We cannot use the same properties for the actual performance because the domain is different and as incompatible as a vegan pizza and a merguez pizza :) for the actual performance the domain is an **event** whereas for "foreseen" performance or casting the domain is E29 Design or Procedure which is a **conceptual object**.
I hope this makes sense.

**Comment by rtroncy** [comment]
> I hope this makes sense.

No. You don't duplicate properties because you find their signature too constrained, and in particular, if they have the duplicate properties have the same intended semantics. You better relax the signature (domain and range) of the properties so that they can be applied to more types. You can also simply declare a domain or a range with a union

**Comment by pierrechoffe** [comment]
@rtroncy 
>You better relax the signature (domain and range) of the properties so that they can be applied to more types

Yep this is what I did (after discussing with @pasqLisena ) for the _foreseen properties_, which are now unique, and used both for the "intended casting" and for the "foreseen performance".

Regarding the _actual performance_, we should respect the CRM/FRBRoo rules and semantics. Not only are there specific naming conventions (like using the past tense for any event-linked property, e.g. _p16 used specific object_) but also we extend classes and properties specific to the events (F31_Performance, *p125_used_object_of_type*, etc.), which therefore cannot be used in another context.

Furthermore, the properties do not have the same semantics. On the one side you have intentions, on the other side you have realisations. On the one side ideas/concepts, on the other side events. Sure, they apply to the same object (e.g. MOP), but this is pretty much all they have in common.

Last but not least, if we want our model to be validated by IFLA and CRM we ought to follow as much as possible their rules. I am quite certain CRM would frown on a "3 in 1" solution.

So my suggestion is to use the properties listed in columns D & E of the [speadsheet](https://docs.google.com/spreadsheets/d/10Oe34ZSIl3bJEVO8O1pq_NoTUuEkZpU3yd5-UMes9vs/edit#gid=0).

**Comment by pasqLisena** [comment]
Closed in dd62889d3627e4e693a2cf176b1f2dcc6f58bdf8 .
